src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
+    const std::unique_ptr<msgpack::zone>& get_zone() const;
```

Comment:
Actually, maybe you even want `msgpack::zone& get_zone() const;` and save the deferencing star that you're currently adding everywhere. Of course this would crash, but so would the current code if the zone pointer is null.